### PR TITLE
Fix build failure on macOS

### LINKED ISSUE #219: Build fails on MacOS 13.2.1: libcli.a is incorrectly linked against libutil.a
The build fails with:
```
g++ -std=c++11 -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -I/opt/local/include -g -O2  -D_FORTIFY_SOURCE=2 -Wall -Wcast-qual -Wextra -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wswitch -Wwrite-strings -g -Werror -Wabi -Wctor-dtor-privacy -Wno-deprecated -Wnon-virtual-dtor -Woverloaded-virtual -Wreorder -Wsign-promo -Wsynth   -o kyua kyua-main.o libcli.a libdrivers.a libengine.a libstore.a libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libstore.a libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libengine.a libstore.a libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libmodel.a libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 libutils.a -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -llua -lm -llutok -L/opt/local/lib -lsqlite3 
ld: in libcli.a(libutils.a), archive member 'libutils.a' with length 12546016 is not mach-o or llvm bitcode file 'libcli.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Removing `libcli_a_LIBADD = libutils.a` from `cli/Makefile.am.inc` allows the build to succeed.

### PR DESCRIPTION
**Pull request by mchoo7**
Prior to this change, building with Homebrew + clang on MacOS would result in a link failure similar to the following:
```
ld: in libcli.a(libutils.a), archive member 'libutils.a' with length 12546016 is not mach-o or llvm bitcode file 'libcli.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Using libtool instead of relying on GNU make's implicit rules for static libraries is the right thing with Homebrew on MacOS as libtool provides the appropriate linker flags for building kyua's shared objects (which are different from how statically linked objects are compiled/linked on MacOS with autotools).

This makes this package consistent with how atf and lutok are built on MacOS.

Closes: #219

### CODE REVIEW

**Review by ngie-eign — CHANGES_REQUESTED** [review]
A number of autogenerated files are being committed as part of this change that should not be committed:
- `admin/config.guess`
- `admin/config.sub`
- `admin/ltmain.sh`

Getting to the change at hand... what's the issue that you're running into MacOS? That part isn't clear from the proposed change (your solution involves using libtool, but why are you proposing adding libtool?).
Sidenote: your `--prefix` isn't standard for (some versions of?) Homebrew; this is part of the reason why these generated files should not be committed to the source tree.

**Comment by ngie-eign** [comment]
I looked a bit closer and the issue provides more detail, as well as the commit messages themselves.. but it was missing from the PR description. Providing this context next time in the PR summary would make it easier for me to understand (without having to dive too deep into the details) what is being done.

_(I'm partly dropping this concern, because PR descriptions are only really maintained in the GH UI/ecosystem)_

The autogenerated files should still be removed.

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,69 @@
...
+/configure
+/configure.scan
+/depcomp
+/install-sh
+/missing
+/stamp-h1
+
+# https://www.gnu.org/software/libtool/
+
+/ltmain.sh
```
Comment (ngie-eign) [inline .gitignore]:
```suggestion
ltmain.sh
```

File: .gitignore [hunk 2]
```diff
@@ -1,23 +1,69 @@
...
+/autoscan.log
+/autoscan-*.log
+/aclocal.m4
+/compile
+/config.cache
+/config.guess
+/config.h.in
+/config.log
+/config.status
+/config.sub
```
Comment (ngie-eign) [inline .gitignore]:
```suggestion
config.sub
```

File: .gitignore [hunk 3]
```diff
@@ -1,23 +1,69 @@
...
+
+# http://www.gnu.org/software/autoconf
+
+autom4te.cache
+/autoscan.log
+/autoscan-*.log
+/aclocal.m4
+/compile
+/config.cache
+/config.guess
```
Comment (ngie-eign) [inline .gitignore]:
```suggestion
config.guess
```

File: configure.ac [hunk 4]
```diff
@@ -42,6 +42,7 @@ AC_CONFIG_TESTDIR([bootstrap])
 
 AM_INIT_AUTOMAKE([1.9 foreign subdir-objects -Wall])
 
+LT_INIT
```
Comment (ngie-eign) [inline configure.ac]: This diff needs to be applied to mute complaints from autotools: https://gist.github.com/ngie-eign/882b78bc57a6b4e50ad224a6394df5e1 . Could you please add this as a [fixup commit](https://mikulskibartosz.name/git-fixup-explained), squash it along with the other changes I've added to your branch, and rebase with the latest version of master?

**Review by ngie-eign — APPROVED** [review]

**Comment by ihoro** [comment]
Hi @fel1x-developer . Could you please share the way you build it for macOS? I would like to repeat the build for testing of recent changes in Kyua. The list of steps, dependencies, versions, etc -- anything will be useful. For instance, I use `atf` installed via `homebrew`.

My motivation: I see that FreeBSD based build is broken (I'm not sure yet whether it's my local issue or due to recent change in *.ac *.am), I would like to fix it and make sure that macOS based build still is fine. And I would like to keep macOS based build tested during the future changes of Kyua.

**Comment by mchoo7** [comment]
> Hi @fel1x-developer . Could you please share the way you build it for macOS? I would like to repeat the build for testing of recent changes in Kyua. The list of steps, dependencies, versions, etc -- anything will be useful. For instance, I use `atf` installed via `homebrew`.
> 
> My motivation: I see that FreeBSD based build is broken (I'm not sure yet whether it's my local issue or due to recent change in *.ac *.am), I would like to fix it and make sure that macOS based build still is fine. And I would like to keep macOS based build tested during the future changes of Kyua.

Hi @ihoro !

First install `lutok`, `atf`, and `lua` with homebrew. Then, build lutok and atf in `~/local` with
```shell
$ autoreconf -i -s -I ~/local/share/aclocal
$ ./configure --prefix ~/local
$ make -j4
$ make install
```

Using the same method, you can build kyua at `~/local` (Don't forget `autoreconf` and `./configure` options)

Feel free to ask me if you have any question!

**Comment by ihoro** [comment]
Thanks for the information. It looks we do it almost the same way. The only diff is that I was using atf as is from `homebrew` without explicit build. I guess it should not be a significant difference.

By the way, from the end user perspective, you use a custom build for your production/personal needs because of:
a) custom patches
b) outdated version from `homebrew` without recent fixes
c) other reasons?

**Comment by mchoo7** [comment]
Mainly b). New versions of `lutok`, `atf`, and `kyua` should be released ASAP to make them installable and executable from Homebrew. I will work on updating those three formulas when new versions of lutok, atf, and kyua come out.

**Comment by ihoro** [comment]
Thank you very much for the outline of your use cases. And kudos for the homebrew formulas maintenance. I have hopes that Kyua could get a new version this summer.

**Pull request merged**